File: LATIHAN1DPBO2023/PHP/CRUD.php
<!-- Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 1
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin. -->

<?php

//deklarasi class CRUD
class CRUD
{
	//constructor
	public function __construct()
	{

	}

	//method untuk menambahkan data (create)
	public function add(&$list, $mhs, $idx2, $name, $nim, $major, $faculty, $image)
	{
		//set data dari paramater dan masukan ke array
		$mhs[$idx2]->setName($name);
		$mhs[$idx2]->setNim($nim);
		$mhs[$idx2]->setMajor($major);
		$mhs[$idx2]->setFaculty($faculty);
		$mhs[$idx2]->setImage($image);

		$list[$idx2] = $mhs[$idx2];
	}

	//method untuk mengubah data (update)
	public function update(&$list, $nimGanti, $name, $nim, $major, $faculty, $image)
	{
		//perulangan untuk mencari data (berdasarkan nim)
		foreach($list as $llist)
		{
			//jika ketemu maka set dengan data baru dari parameter
			if($llist->getNim() == $nimGanti)
			{
				$llist->setName($name);
				$llist->setNim($nim);
				$llist->setMajor($major);
				$llist->setFaculty($faculty);
				$llist->setImage($image);
			}
		}
	}

	//method untuk menghapus data (delete)
	public function delete(&$list, $nimHapus)
	{
		$i;
		//perulangan untuk mencari data (berdasarkan nim)
		for($i = 0; $i < count($list); $i++)
		{
			if($list[$i]->getNim() == $nimHapus)
			{
				array_splice($list, $i, 1);
			}
		}
	}

	//method untuk menampilkan data (read)
	public function show($list)
	{
		//output berupa tabel
		echo "<table border='1' cellspacing='0' cellpading='10' cellpadding='10'";
		echo "<tr>";
		echo "<th>No</th>";
		echo "<th>Nama</th>";
		echo "<th>NIM</th>";
		echo "<th>Program Studi</th>";
		echo "<th>Fakultas</th>";
		echo "<th>Foto Profile</th>";
		echo "</tr>";
		$j = 1;
		$i;
		for($i = 0; $i < count($list); $i++)
		{
			echo "<tr>";
			echo  "<td>" . $j . '. ' . "</td>";
			echo "<td>" . $list[$i]->getName() . "</td>";
			echo "<td>" . $list[$i]->getNim() . "</td>";
			echo "<td>" . $list[$i]->getMajor() . "</td>";
			echo "<td>" . $list[$i]->getFaculty() . "</td>";
			echo "<td>" . "<img src='" . $list[$i]->getImage() . "' style='width: 100px;'>" . '</td>';
			echo "</tr>";
			$j = $j + 1;
		}
		echo "</table>";
		echo "<br>";
	}

	//destructor
	public function __destruct()
	{
		
	}
}

?>

File: LATIHAN1DPBO2023/CPP/CRUD.cpp
/*Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 1
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//library yang digunakan
#include <bits/stdc++.h>
#include "Mahasiswa.cpp"

using namespace std;

//deklarasi class CRUD
class CRUD
{
	public:
		//atribut untuk tampungan
		string name;
		string nim;
		string major;
		string faculty;
		int j;

		//constructor (dibuat kosong karena tidak diperlukan untuk diisi)
		CRUD()
		{

		}

		//method untuk menambahkan data (create)
		void add(list<Mahasiswa> *llist)
		{
			//instance object menggunakan class Mahasiswa
			Mahasiswa temp;
			j = 0;

			//inputan untuk data mahasiswa
			cout << "Masukkan Nama:";
			cin >> name;
			cout << "Masukkan NIM:";
			cin >> nim;
			cout << "Masukkan Program Studi:";
			cin >> major;
			cout << "Masukkan Fakultas:";
			cin >> faculty;

			//perulangan sebanyak data dalam list
			for(list<Mahasiswa>::iterator it = (*llist).begin(); it != (*llist).end(); it++)
			{
				//pengecekan jika nim yang diinput sama dengan nim pada list
				//nim digunakan sebagai primary key
				if(nim == it->getNim())
				{
					//print warning bahwa data sudah tersedia
					//variabel j digunakan sebagai penanda
					j = 1;
					cout << "\nTidak dapat menambahkan data, NIM sudah tersedia!" << '\n';
				}
			}

			//jika variabel j tetap nol (artinya tidak ada nim yang sama / data belum tersedia)
			if(j == 0)
			{
				//set data inputan ke object tampungan yang sebelumnya telah dibuat
				temp.setName(name);
				temp.setNim(nim);
				temp.setMajor(major);
				temp.setFaculty(faculty);

				//push object tersebut ke dalam list
				(*llist).push_back(temp);
				cout << "\nData berhasil ditambahkan!" << '\n';
			}
		}

		//method untuk mengubah data (update)
		void update(string nim, list<Mahasiswa> *llist)
		{
			//variabel penanda
			j = 0;

			//perulangan sebanyak data pada list
			for(list<Mahasiswa>::iterator it = (*llist).begin(); it != (*llist).end(); it++)
			{
				//jika nim pada parameter (yaitu nim yang akan diubah) sama dengan nim pada list
				if(nim == it->getNim())
				{
					//inputan untuk data yang akan diubah
					cout << "Masukkan Nama:";
					cin >> name;
					cout << "Masukkan NIM:";
					cin >> nim;
					cout << "Masukkan Program Studi:";
					cin >> major;
					cout << "Masukkan Fakultas:";
					cin >> faculty;

					//set inputan
					it->setName(name);
					it->setNim(nim);
					it->setMajor(major);
					it->setFaculty(faculty);

					cout << "\nData berhasil diubah!" << '\n';

					j = 1;
				}
			}

			//jika variabel j tetap nol (artinya data yang akan diupdate tidak tersedia)
			if(j == 0)
			{
				//print warning bahwa data tidak tersedia
				cout << "\nData tidak tersedia!" << '\n';
			}
		}

		//method untuk menghapus data (delete)
		void del(string nim, list<Mahasiswa> *llist)
		{
			j = 0;

			//perulangan sebanyak data pada list
			for(list<Mahasiswa>::iterator it = (*llist).begin(); it != (*llist).end(); it++)
			{
				//jika nim dari parameter sama dengan nim pada list
				if(nim == it->getNim())
				{
					//hapus data pada list
					it = (*llist).erase(it);
					it--;
					cout << "\nData berhasil dihapus!" << '\n';
					j = 1;
				}
			}

			//jika data tidak tersedia
			if(j == 0)
			{
				cout << "\nData tidak tersedia!" << '\n';
			}
		}

		//method untuk menampilkan data (read)
		void show(list<Mahasiswa> *llist)
		{
			//output dalam bentuk tabel
			cout << "\nList Mahasiswa:\n";
			int i = 0;
			cout << "+=======================================+\n" << "|  Nama  |  NIM  |  Prodi  |  Fakultas  |\n" << "+=======================================+\n";
			for(list<Mahasiswa>::iterator it = (*llist).begin(); it != (*llist).end(); it++)
			{
				cout << "| " <<(i + 1) << ". " << it->getName() << " | " << it->getNim() << " | " << it->getMajor() << " | " << it->getFaculty() << " |\n";
				cout << "+---------------------------------------+\n";
				i++;
			}
		}

		//destructor
		~CRUD()
		{

		}
};